### PR TITLE
implementing local storage for manifest links

### LINKED ISSUE #15: Implement Local Storage for Manifest Links
**Description**: 
As a user,
I need the ability to store and retrieve manifest links in local storage
So that the manifest links persist across page refreshes.

Create functionality to store and retrieve manifest links in local storage, allowing persistence across page refreshes.

**Acceptance Criteria**:

- [ ] - When a user inputs a manifest link it should be saved in local storage.
- [ ] - On page reload the manifest link should still be accessible.

### PR DESCRIPTION
**Pull request by haneenalsewari**
Fixes #15
The ability to store and retrieve manifest links using local storage has been implemented. Now, when a user inputs a manifest link, it persists across page reloads by saving it in the browser's local storage.
This change was necessary to ensure users don't lose their manifest links upon page refresh or closure, enhancing the usability of the tool by remembering past inputs
A new manifestStorage.js file was created to handle saving and retrieving manifest links from local storage. The UI was also updated in tools.html to allow users to input and save a manifest link. Upon page load, any saved manifest links will be retrieved and displayed.

### CODE REVIEW

File: web/js/manifestStorage.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const MANIFEST_LINKS_KEY ='storedManifestLinks';
+
+/**
+ * Save the given manifest link to local storage.
+ */
+export function storeManifestLink(manifestLink) {
+    let manifestLinks = getStoredManifestLinks();
+
+    // avoid duplicates, add new manifest link if not already present
```
Comment (haydenkarl22) [inline web/js/manifestStorage.js:9]: I don't think this comment is necessary, id recommend removing it.

File: web/js/manifestStorage.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    if (!manifestLinks.includes(manifestLink)) {
+        manifestLinks.push(manifestLink);
+    }
+
+    // save updated links array to local storage
+    localStorage.setItem(MANIFEST_LINKS_KEY, JSON.stringify(manifestLinks));
+}
+
+/**
+ * Retrieve the stored manifest links from local storage
```
Comment (haydenkarl22) [inline web/js/manifestStorage.js:19]: this comment is also not necessary, id also remove this one.

**Review by alar12 — CHANGES_REQUESTED** [review]
I see the functionality is implemented well but need to link with @haydenkarl22 #17. As it is merged in main go ahead and pull the changes from main and link the functionality. Also render the url's in local storage below the url box.

**Review by alar12 — APPROVED** [review]
The functionality is working well as per requirement good work!!.

**Pull request merged**